### PR TITLE
vkjson: Add whole-instance support

### PR DESCRIPTION
**Pull request by critsec**
Previously vkjson only represented, imported, and exported a single VkPhysicalDevice at a time. Now it can also handle an entire instance: instance extensions, instance layers (and their extensions), as well as a set of devices.

This change remains source compatible, and the vkjson_info command takes the same arguments and produces the same output as before; whole-instance is a new mode. It is not binary compatible, but a simple recompile of the client will work; binary compatibility could be retained easily, but since it's intended to be used as a static rather than shared library this didn't seem important.

### CODE REVIEW

File: libs/vkjson/vkjson_instance.cc [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+      nullptr};
+  VkInstance vkinstance;
+  result = vkCreateInstance(&instance_info, nullptr, &vkinstance);
+  if (result != VK_SUCCESS)
+    return VkJsonInstance();
+
+  count = 0;
+  result = vkEnumeratePhysicalDevices(vkinstance, &count, nullptr);
+  if (result != VK_SUCCESS)
+    return VkJsonInstance();
```
Comment (pimanttr) [inline libs/vkjson/vkjson_instance.cc]: Destroy instance in this case - otherwise it leaks.

File: libs/vkjson/vkjson_instance.cc [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+  std::vector<VkPhysicalDevice> devices(count, VK_NULL_HANDLE);
+  result = vkEnumeratePhysicalDevices(vkinstance, &count, devices.data());
+  if (result != VK_SUCCESS)
+    return VkJsonInstance();
```
Comment (pimanttr) [inline libs/vkjson/vkjson_instance.cc]: ditto.

**Comment by pimanttr** [comment]
LGTM after couple of issues fixed

**Comment by critsec** [comment]
Missing vkDestroyInstance fixed; thanks Antoine.

Jon, I don't have a Windows machine to check whether this builds. Anyone there who can check?

**Comment by jon-lunarg** [comment]
Runs on Windows just fine even with multiple ICDs. Cherry picked commits to master branch. Thanks!

**Pull request closed without merging**